test test_no_config fails with custom ES port

On gentoo linux we run the curator tests with es on custom port. Test test_no_config fails because without config it tries to connect to port 9200. 

How to reproduce:
start ES on port 15000

```
export TEST_ES_SERVER="localhost:15000"
python setup.py test
```

```
...
test_no_config (test.integration.test_cli.TestCLIMethods) ... FAIL
...
======================================================================
FAIL: test_no_config (test.integration.test_cli.TestCLIMethods)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/curator-user/curator-4.2.4/test/integration/test_cli.py", line 57, in test_no_config
    self.assertEqual(0, result.exit_code)
AssertionError: 0 != -1
-------------------- >> begin captured logging << --------------------
DEBUG curator.validators.SchemaCheck               __init__:26   Schema: <Schema({'client': {'port': Any([None, All(Coerce(int, msg=None), Range(min=1, max=65535, min_included=True, max_included=True, msg=None), msg=None)]), 'url_prefix': Any([None, <type 'str'>, <type 'unicode'>]), 'aws_key': Any([None, <type 'str'>, <type 'unicode'>]), 'client_cert': Any([None, <type 'str'>, <type 'unicode'>]), 'ssl_no_validate': <function Boolean at 0x7f6c66fa3410>, 'certificate': Any([None, <type 'str'>, <type 'unicode'>]), 'use_ssl': <function Boolean at 0x7f6c66f6b6e0>, 'hosts': Any([None, <type 'str'>, <type 'unicode'>, <type 'list'>]), 'http_auth': Any([None, <type 'str'>, <type 'unicode'>]), 'aws_region': Any([None, <type 'str'>, <type 'unicode'>]), 'client_key': Any([None, <type 'str'>, <type 'unicode'>]), 'aws_secret_key': Any([None, <type 'str'>, <type 'unicode'>]), 'master_only': <function Boolean at 0x7f6c66fa3668>, 'timeout': All(Coerce(int, msg=None), Range(min=1, max=86400, min_included=True, max_included=True, msg=None), msg=None)}, 'logging': {'blacklist': Any([None, <type 'list'>]), 'loglevel': Any([None, 'NOTSET', 'DEBUG', 'INFO', 'WARNING', 'ERROR', 'CRITICAL', All(Coerce(int, msg=None), Any([0, 10, 20, 30, 40, 50]), msg=None)]), 'logfile': Any([None, <type 'str'>, <type 'unicode'>]), 'logformat': Any([None, All(Any([<type 'str'>, <type 'unicode'>]), Any(['default', 'json', 'logstash']), msg=None)])}}, extra=PREVENT_EXTRA, required=False) object at 0x7f6c66fa9e10>
DEBUG curator.validators.SchemaCheck               __init__:27   "Client Configuration" config: {'client': {}, 'logging': {}}
INFO            curator.cli                    cli:147  Action ID: 1: "close" not performed because "disable_action" is set to True
INFO            curator.cli                    cli:151  Preparing Action ID: 2, "delete_indices"
--------------------- >> end captured logging << ---------------------
```

Curator tries to connect to port 9200
```
# tcpdump -i lo -n -A -s 2000 'port 9200'
dropped privs to tcpdump
tcpdump: verbose output suppressed, use -v or -vv for full protocol decode
listening on lo, link-type EN10MB (Ethernet), capture size 2000 bytes
07:20:52.528046 IP 127.0.0.1.45888 > 127.0.0.1.9200: Flags [S], seq 247319827, win 43690, options [mss 65495,sackOK,TS val 6145446 ecr 0,nop,wscale 7], length 0
E..<..@.@.4..........@#..............0.........
.]..........
07:20:52.528067 IP 127.0.0.1.9200 > 127.0.0.1.45888: Flags [R.], seq 0, ack 247319828, win 0, length 0
E..(..@.@.].........#..@........P.......
07:20:52.529800 IP 127.0.0.1.45890 > 127.0.0.1.9200: Flags [S], seq 779689414, win 43690, options [mss 65495,sackOK,TS val 6145448 ecr 0,nop,wscale 7], length 0
E..<.n@.@.dK.........B#..y...........0.........
.]..........
07:20:52.529815 IP 127.0.0.1.9200 > 127.0.0.1.45890: Flags [R.], seq 0, ack 779689415, win 0, length 0
E..(..@.@.].........#..B.....y..P....[..
07:20:52.531499 IP 127.0.0.1.45892 > 127.0.0.1.9200: Flags [S], seq 3606032370, win 43690, options [mss 65495,sackOK,TS val 6145450 ecr 0,nop,wscale 7], length 0
E..<jX@.@..a.........D#..............0.........
.]..........
07:20:52.531514 IP 127.0.0.1.9200 > 127.0.0.1.45892: Flags [R.], seq 0, ack 3606032371, win 0, length 0
E..(..@.@.].........#..D........P...S...
07:20:52.533164 IP 127.0.0.1.45894 > 127.0.0.1.9200: Flags [S], seq 1617695886, win 43690, options [mss 65495,sackOK,TS val 6145451 ecr 0,nop,wscale 7], length 0
E..<.%@.@.k..........F#.`l...........0.........
.]..........
07:20:52.533178 IP 127.0.0.1.9200 > 127.0.0.1.45894: Flags [R.], seq 0, ack 1617695887, win 0, length 0
E..(..@.@.].........#..F....`l..P...e...
```

--- a/docs/Changelog.rst
+++ b/docs/Changelog.rst
@@ -9,7 +9,12 @@ Changelog
 **General**
 
   * Add and increment test versions for Travis CI. #839 (untergeek)
-  
+
+**Bug Fixes**
+
+  * Fix cli integration test when different host/port are specified.  Reported
+    in #843 (untergeek)
+
 4.2.4 (7 December 2016)
 -----------------------
 

--- a/test/integration/test_cli.py
+++ b/test/integration/test_cli.py
@@ -37,6 +37,15 @@ class TestCLIMethods(CuratorTestCase):
                     )
         self.assertEqual(-1, result.exit_code)
     def test_no_config(self):
+        # This test checks whether localhost:9200 is provided if no hosts or
+        # port are in the configuration. But in testing, sometimes
+        # TEST_ES_SERVER is set to something other than localhost:9200.  In this
+        # case, the test here would fail.  The if statement at the end now
+        # compensates. See https://github.com/elastic/curator/issues/843
+        localtest = False
+        if (host == 'localhost' or host == '127.0.0.1') and \
+          port == 9200:
+            localtest = True
         self.create_indices(10)
         self.write_config(
             self.args['configfile'],
@@ -54,7 +63,10 @@ class TestCLIMethods(CuratorTestCase):
 
                     ],
                     )
-        self.assertEqual(0, result.exit_code)
+        if localtest:
+            self.assertEqual(0, result.exit_code)
+        else:
+            self.assertEqual(-1, result.exit_code)
     def test_no_logging_config(self):
         self.create_indices(10)
         self.write_config(

